lib/improver/tests/utilities/test_temporal.py

```diff
@@ -429,5 +446,92 @@ def test_invalid_date_format(self):
                                forecast_time=6)
 
 
+class Test_unify_forecast_reference_time(IrisTest):
+
+    """Test the unify_forecast_reference_time function."""
+
+    def setUp(self):
+        """Set up a UK deterministic cube for testing."""
+        cube_uk_det = add_model_id_and_model_configuration(
+            set_up_temperature_cube(timesteps=3), model_ids=[1000],
+            model_configurations=["uk_det"], promote_to_new_axis=True)
+        self.cube_uk_det = add_forecast_reference_time_and_forecast_period(
+            cube_uk_det, time_point=[412233.0, 412235.0, 412237.0],
+            fp_point=[6., 8., 10.])
+
+    def test_cubelist_input(self):
+        """Test when supplying a cubelist as input containing cubes
+        representing UK deterministic and UK ensemble model configuration
+        and unifying the forecast_reference_time, so that both model
+        configurations have a common forecast_reference_time."""
+        cube_uk_ens = add_model_id_and_model_configuration(
+            set_up_temperature_cube(timesteps=3), model_ids=[2000],
+            model_configurations=["uk_ens"], promote_to_new_axis=True)
+        cube_uk_ens = add_forecast_reference_time_and_forecast_period(
+            cube_uk_ens, time_point=[412231.0, 412233.0, 412235.0],
+            fp_point=[4., 6., 8.])
```

Comment:
I've modified the fp_points, which changes the forecast_reference_time generated by this function so that the test does do what we want. The result remains unchanged (correctly).